library/std/src/net/ip.rs

```diff
@@ -958,18 +951,21 @@ impl PartialOrd<IpAddr> for Ipv4Addr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Ord for Ipv4Addr {
     fn cmp(&self, other: &Ipv4Addr) -> Ordering {
-        u32::from_be(self.inner.s_addr).cmp(&u32::from_be(other.inner.s_addr))
+        // We try to compare as native endian `u32` by using `u32::from_be`.
+        // In LE target, it swap bytes of BE inner to make it as LE `u32`.
+        // In BE target, it does nothing so we have BE `u32`.
+        u32::from_be(self.inner.to_be()).cmp(&u32::from_be(other.inner.to_be()))
     }
 }
 
-impl AsInner<c::in_addr> for Ipv4Addr {
-    fn as_inner(&self) -> &c::in_addr {
-        &self.inner
+impl IntoInner<c::in_addr> for Ipv4Addr {
+    fn into_inner(self) -> c::in_addr {
+        c::in_addr { s_addr: self.inner.to_be() }
```

Comment:
Since `c::in_addr` is not a real inner type in Ipv4Addr. I don't think this trait impl has correct semantic.
But and `impl From<Ipv4Addr> for c::in_addr` needs a `#[stable]` attribute.
So I think about making a unstable method on Ipv4Addr to do this conversion.